FirebaseMessaging/Sources/FIRMessaging.m

```diff
@@ -838,11 +842,12 @@ - (void)defaultInstanceIDTokenWasRefreshed:(NSNotification *)notification {
   }
   // Retrieve the Instance ID default token, and should notify delegate and
   // trigger notification as long as the token is different from previous state.
-  NSString *oldToken = self.defaultFcmToken;
-  self.defaultFcmToken = [(NSString *)notification.object copy];
-  if ((self.defaultFcmToken.length && oldToken.length &&
-       ![self.defaultFcmToken isEqualToString:oldToken]) ||
-      self.defaultFcmToken.length != oldToken.length) {
+  NSString *oldToken = _tokenManager.defaultFCMToken;
+  NSString *newToken = [(NSString *)notification.object copy];
+  if ((newToken.length && oldToken.length && ![newToken isEqualToString:oldToken]) ||
```

Comment:
optional: I took me some time to understand this check. I wonder would it be more readable if we explicitly check new and old token for `nil` like:
```
(newToken != nil & oldToken != nil & ![newToken isEqualToString:oldToken])
```

Reply:
what happen if old token is nil and new token is a string, we want this case to be true.
Or you only mean the first line of the condtion?

Reply:
Yeah, I mean keeping the same checks but replace `token.length` by `token != nil` which reads a bit better to me. I don't really have a strong opinion here. It looks fine especially after moving the checks to the separate method.